idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1497 @@
+                              self.liquid_phase.length_domain,
+                              initialize=0.001,
+                              doc='Volumetric liquid holdup [-]')
+
+        def rule_holdup_vap(blk, t, x):
+            return blk.eps_ref - blk.holdup_liq[t, x]
+
+        self.holdup_vap = Expression(self.flowsheet().time,
+                                     self.vapor_phase.length_domain,
+                                     rule=rule_holdup_vap,
+                                     doc='Volumetric vapor holdup [-]')
+        
+        # Column pressure profile
+        if not self.config.vapor_phase.has_pressure_change:
+            self.column_pressure = Var(
+                initialize=107650,
+                units=pyunits.Pa,
+                doc='Fixed operating pressure of column')
+        else:                
+            def rule_deltaP(blk, t, x):
+                return blk.vapor_phase.deltaP[t, x] == self.config.column_pressure_drop
+                    
+            self.deltaP_constr = Constraint(self.flowsheet().time,
+                                 self.vapor_phase.length_domain,
+                                 rule=rule_deltaP,
+                                 doc='Irrigated packing pressure drop (per unit length)'
+                                 'constraint with user defined value or expression for deltaP')
+        
+        # Calculating gas velocity at flooding point
+        def rule_flood_velocity(blk, t, x):
+            x_liq = blk.liquid_phase.length_domain.at(blk.zi[x].value)
+            dens_liq = blk.liquid_phase.properties[t, x_liq].mw/\
+                    blk.liquid_phase.properties[t, x_liq].vol_mol_phase['Liq']
+                    
+            dens_vap = blk.vapor_phase.properties[t, x].mw/\
+                    blk.vapor_phase.properties[t, x].vol_mol_phase['Vap']
+                    
+            H = (blk.liquid_phase.properties[t, x_liq].flow_mass_phase['Liq']/\
+                blk.vapor_phase.properties[t, x].flow_mass_phase['Vap'])*\
+                (dens_liq/dens_vap)**0.5
+                
+            mu_water = 0.001*pyunits.Pa*pyunits.s
+            
+            gref = pyunits.convert(g_ref, to_units=pyunits.ft/(pyunits.s)**2)
+            
+            aref = pyunits.convert(self.a_ref, to_units=((pyunits.ft)**2)/(pyunits.ft)**3)
+            
+            return ((gref*((self.eps_ref)**2)/aref)*(dens_liq/dens_vap)*\
+                ((blk.liquid_phase.properties[t, x_liq].visc_d_phase['Liq']/mu_water)*(-0.2))*\
+                    exp(-4*(H)**(0.25)))**0.5
+                
+        self.gas_velocity_flood = Expression(self.flowsheet().time,
+                                self.vapor_phase.length_domain,
+                                rule=rule_flood_velocity,
+                                doc='Gas velocity at flooding point')
+            
+        # Flooding fraction 
+        def rule_flood_fraction(blk, t, x):
```

Comment:
We need to apply this rule to construct the flood_fraction `Constraint` to allow for the calculation of the required vapor velocity to determine the column diameter during design. Hence `self.flood_fraction` should be a  variable. So during simulation, the `flood_fraction` is calculated and for design purposes, it is fixed as desired by the user to determine the corresponding velocity and then the diameter of the column from the admissible flows.

Reply:
This is probably the best way to do this.